aip/auth/4122.md

```diff
@@ -0,0 +1,349 @@
+  ],
+  "audience": "audience",
+  "includeEmail": "true|false"
+}
+```
+
+The following is an example of the payload for a direct request for an id token
+with email included:
+
+```json
+{
+  "audience": [
+    "example-sa@example-project.iam.gserviceaccount.com"
+  ],
+  "includeEmail": true 
+}
+```
+
+The following is an example of the payload for a delegated request for an id 
+token with email excluded:
+
+```json
+{
+  "delegates": [
+    "sa-delegate1@project-id.iam.gserviceaccount.com",
+    "sa-delegate2@project-id.iam.gserviceaccount.com"
+  ],
+  "audience": "example-sa@example-project.iam.gserviceaccount.com"
+}
+```
+
+### Creating a self-signed JSON Web Token (JWT)
+
+#### Parameters
+
+The auth library **must** implement the language-specific abstractions required
+to allow a caller to specify the following parameters for a request to create a
+self signed JWT on behalf of a service account.
+
+**Source Credentials** - **Required**
+
+The library specific representation of the service/user account requesting
+ the impersonation.
+ 
+**Target Principal** - **Required**
+
+The email address of the service account to impersonate, e.g. 
+`sa-name@project.iam.gserviceaccount.com`.
+
+**JWT Payload** - **Required**
+
+A JSON object that contains the JWT Claims Set to sign. If the JWT Claims set
+contains an `exp` claim and the JWT is used to call a Google API, auth libraries
+**must** validate that the `exp` claim is set to no more that 1 hour in the future.
+
+**Delegates** - **Optional**
+
+An ordered list of service account emails in the delegation chain, if used.
+
+#### Expected Behavior
+
+Auth libraries **must** send a POST request to the 
+[projects.serviceAccounts.signJwt][4] method, using an access
+token generated for the Source Credential to authorize the request:
+
+```
+POST https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/<TargetPrincipal>:signJWT
+```
+
+The request JSON body **must** be of the form:
+
+```json
+{
+  "delegates": [
+    // optional list of delegates
+  ],
+  "payload": "{ // JWT Claims set }"
+}
+```
+
+If the request is successful, the response body contains a signed JWT and the 
+signing key ID that was used to sign the JWT, as in the following example:
+
+```json
+{
+  "keyId": "42ba1e...fc0a",
+  "signedJwt": "eyJ0eXAi...NiJ9"
+}
+```
+
+Auth libraries **must** return both to the caller.
+
+### Creating a self-signed blob
+
+#### Parameters
+
+The auth library **must** implement the language-specific abstractions required
+to allow a caller to specify the following parameters for a request to create a
+self signed blob on behalf of a service account.
+
+**Source Credentials** - **Required**
+
+The library specific representation of the service/user account requesting
+ the impersonation.
+ 
+**Target Principal** - **Required**
+
+The email address of the service account to impersonate, e.g. 
+`sa-name@project.iam.gserviceaccount.com`.
+
+**Payload** - **Required**
+
```

Comment:
In python and Java libraries, user just passes in bytes, then the library base64 encodes the bytes and use that as the `payload`. So something like `BytesToSign` seems to be more proper than `Payload`, also it could be any bytes instead of JSON related object.